front/src/modules/scenario/components/AddOrEditScenarioModal.tsx

```diff
@@ -25,29 +25,43 @@ import ModalHeaderSNCF from 'common/BootstrapSNCF/ModalSNCF/ModalHeaderSNCF';
 import { ModalContext } from 'common/BootstrapSNCF/ModalSNCF/ModalProvider';
 import { InfraSelectorModal } from 'modules/infra/components/InfraSelector';
 import { setFailure, setSuccess } from 'reducers/main';
+import { ConfirmModal } from 'common/BootstrapSNCF/ModalSNCF';
+import useInputChange from 'utils/hooks/useInputChange';
+import useOutsideClick from 'utils/hooks/useOutsideClick';
 
-type CreateOrPatchScenarioForm = ScenarioPatchForm & {
+type ScenarioForm = ScenarioPatchForm & {
   id?: number;
   infra_id?: number;
   electrical_profile_set_id?: number | null;
 };
 
 type AddOrEditScenarioModalProps = {
   editionMode?: boolean;
-  scenario?: CreateOrPatchScenarioForm;
+  scenario?: ScenarioForm;
 };
 
 type createScenarioParams = {
   projectId: string;
   studyId: string;
 };
 
+const emptyScenario: ScenarioCreateForm = {
+  description: '',
+  electrical_profile_set_id: NaN,
+  infra_id: undefined as unknown as number,
+  name: '',
+  tags: [],
+};
+
```

Comment:
Then you can remove your condition a few lines below. And you will have to declare this const inside the component.

```suggestion
const emptyScenario: ScenarioForm = {
  description: '',
  electrical_profile_set_id: NaN,
  infra_id: infraId ?? undefined,
  name: '',
  tags: [],
};

```

Reply:
fix